src/co_tools/co_fastq.py

```diff
@@ -0,0 +1,156 @@
+
+# local imports
+from .get_logger import LOGGER
+
+
+def get_fastq_pair(dir_path: str = "../data"):
+    total_dirs = 0
+    for base, dirs, files in os.walk(dir_path):
+        for dir in dirs:
+            total_dirs += 1
+    if total_dirs != 2:
+        LOGGER.error(
+            f"The fastq files in {dir_path} are not properly configured"
+            + " to use this function. There should be only 2 folders"
+            + " inside the data folder."
+        )
+        return 0
+    prefix_dict, this_prefix = {}, None
+    fwd, rev = None, None
+    for path in glob(str(f"{dir_path}/**/*.fastq.gz"), recursive=True):
+        if prefix := get_prefix(path):
+            if prefix in prefix_dict:
+                prefix_dict[prefix].append(path)
+                if len(prefix_dict[prefix]) == 3:
+                    LOGGER.info(
+                        f"prefix {prefix} occurs 3 times in the {dir_path} folder"
+                    )
+                    this_prefix = prefix
+                    break
+            else:
+                prefix_dict[prefix] = [path]
+        else:
+            LOGGER.warning(f"No prefix determined for {path}")
+    if not prefix_dict:
+        LOGGER.error(f"No files found in {dir_path}")
+        return 0
+    if not this_prefix:
+        LOGGER.error(f"fastq files in {dir_path} not properly organized")
+        return 0
+    for path in prefix_dict[this_prefix]:
+        if get_read_direction(path) == "1":
+            fwd = path
+        elif get_read_direction(path) == "2":
+            rev = path
+    if fwd and rev:
+        LOGGER.info(f"returning {fwd},{rev}")
+        return f"{fwd},{rev}"
+    else:
+        LOGGER.error(
+            f"Could not find complementary pair of fastq files in {dir_path}"
+        )
+        return 0
+
+
+def get_fwd_fastqs(dir: str = "../data"):
+    if some_fastq := (
+        subprocess.check_output(["find", "-L", dir, "-name", "*.fastq.gz"])
+        .decode("utf-8")
+        .strip()
+    ):
+        if "\n" in some_fastq:
+            some_fastq = some_fastq.split("\n")[0]
+        LOGGER.debug(f"some_fastq: {some_fastq}")
+        pattern = get_read_pattern(some_fastq)
+        LOGGER.debug(f"pattern: {pattern}")
+        subprocess_files = (
+        subprocess.check_output(["find", "-L", dir, "-name", f"*{pattern}"])
+        .decode("utf-8")
+        .strip()
+    )
+        LOGGER.debug(f"subprocess_files:\n{subprocess_files}")
+        subprocess_files = subprocess_files.split("\n")
+        subprocess_files.sort()
+        LOGGER.debug(f"sorted subprocess_files:\n{subprocess_files}")
+        files = ""
+        for elem in files:
+            files += f"{elem}\n"
+        files = files.strip()
+        LOGGER.debug(f"returning files:\n{files}")
+        return files
+    else:
+        LOGGER.error(f"There are no fastq.gz files in the {dir} directory")
+        return 0
+
+
+def get_read_direction(filepath: str):
+    filename = filepath.split("/")[-1]
```

Comment:
Use pathlib, not string splitting.

Reply:
What benefit is there to convert from string to a pathlib object and then convert back to string?

Reply:
Clarity of purpose and (though it doesn't matter for us) interoperability between operating systems.

Reply:
The name of the variable `filename` makes it explicitly clear what the value of the string splitting is intended to be. 
What possible conflict with another operating system does a string input have? Are you suggesting a different operating system might not default to utf-8 encoding?